opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -123,6 +120,8 @@
 
     public WatchDogService watchDog;
 
+    private final Set<OnConfigurationChangedListener> listeners = new CopyOnWriteArraySet<>();
```

Comment:
why CopyOnWriteArraySet ?

Reply:
I'm iterating over the set when the configuration is changed. This could throw https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html if someone would try to register another listener at that time.